### PR TITLE
Issue#9 regarding incorrect results is now resolved.

### PR DESCRIPTION
**Pull request by arushidogra**
Issue#9 regarding incorrect results is now resolved.
Everything is working properly. 
screenshot from 2016-03-08 15 55 34
screenshot from 2016-03-08 15 55 47
screenshot from 2016-03-08 15 56 00

### CODE REVIEW

**Comment by arushidogra** [comment]
@copyninja  , I have done PR for development branch. Can you look into this?

File: soundex/__init__.py [hunk 1]
```diff
@@ -142,12 +142,15 @@ def compare(self, string1, string2):
...
+        if soundex1[1:] == soundex2[1:] and soundex1[0] == soundex2[0]:
             # Strings sound phonetically same
             return 1
 
         # Strings are not same
-        return 2
+        if soundex1[1:] == soundex2[1:] and soundex1[0] != soundex2[0]:
+            return 2
+        else:
+            return -1
```
Comment (copyninja) [inline soundex/__init__.py]: What is the meaning of return -1?..

**Comment by copyninja** [comment]
@arushidogra Please use some meaningful messages in commit. All commit have same messages

File: soundex/__init__.py (hunk 1 above)
Comment (arushidogra) [inline soundex/__init__.py]: When -1 is returned "Words does not match" is displayed.

**Comment by arushidogra** [comment]
@copyninja, Should I make a PR with meaningful commits?

File: soundex/__init__.py (hunk 1 above)
Comment (copyninja) [inline soundex/__init__.py]: Doesn't return 2 do the same?. Why 2 return value for same meaning?.

**Comment by stultus** [comment]
> @copyninja,  Should I make a PR with meaningful messages in commits?
@arushidogra  Yes.  the purpose of version controlling is to go back to a particular state when we want. 
If we cannot pinpoint a particular commit using the commit message, there is no point in that.

File: soundex/__init__.py (hunk 1 above)
Comment (arushidogra) [inline soundex/__init__.py]: Return 2 , phonetically same but different languages. I have explained as comments in the new commits.

**Comment by arushidogra** [comment]
@copyninja I have done the changes.

**Comment by copyninja** [comment]
@arushidogra This is not acceptable reason being we have 3 + 3 commits which are contradicting each other. I would suggest do fresh clone apply changes there also keep note of what @stultus said.

**Comment by arushidogra** [comment]
@copyninja , okay i will do the changes. The new commits are same .. only difference is the comments were not added properly in the previous commits.. I will make a PR again  with the changes.

**Comment by copyninja** [comment]
On Wed, Mar 9, 2016 at 2:29 PM, Arushi Dogra <notifications@github.com> wrote:
>
> @copyninja , okay i will do the changes. The new commits are same .. only difference is the comments were not added properly in the previous commits.. I will make a PR again with the changes.

Actually with your new commits we do not get the feature of reverting
single change as mentioned by @stultus.




-- 

Vasudev Kamath
http://copyninja.info
copyninja@{frndk.de|vasudev.homelinux.net}

**Comment by arushidogra** [comment]
@copyninja, okay I understood. So for that I will do commits for different files differently.. It will solve the issue right?

**Comment by copyninja** [comment]
Closing this PR as you already gave a different PR

**Pull request closed without merging**